api/types/src/convert.rs

```diff
@@ -422,19 +429,32 @@ impl<'a, R: ModuleResolver + ?Sized> MoveConverter<'a, R> {
         key: &[u8],
         value: &[u8],
     ) -> Result<Option<DecodedTableData>> {
-        if !self.db.indexer_enabled() && !self.db.indexer_async_v2_enabled() {
-            return Ok(None);
-        }
-        let table_info = match self.db.get_table_info(handle) {
-            Ok(ti) => ti,
-            Err(_) => {
+        // Define a closure to encapsulate the logic
+        let get_table_info = |handle| -> Result<Option<TableInfo>> {
+            if let Some(table_info_reader) = self.table_info_reader.as_ref() {
+                // If table_info_reader exists, use it to get table_info
+                Ok(table_info_reader.get_table_info(handle)?)
+            } else if self.db.indexer_enabled() {
+                // If indexer is enabled in db, get table_info from the db
+                Ok(Some(self.db.get_table_info(handle)?))
+            } else {
+                // If neither condition is met, return None
+                Ok(None)
+            }
+        };
```

Comment:
cuz now we have two db readers since new indexer db is no longer part of the aptosdb. no matter what, we need to check two sources, but i could def improve on readability